iotfunctions/pipeline.py

```diff
@@ -1122,31 +1122,36 @@ def execute_stage(self, stage, df, start_ts, end_ts):
         # entities to retrieve data from
         entities = self.exec_payload_method(method_name='get_entity_filter', default_output=None, raise_error=False)
         usage = 0
+        kpi_function_name = "unknown"
 
         # There are two possible signatures for the execute method
         try:
             result = stage.execute(df=df, start_ts=start_ts, end_ts=end_ts, entities=entities)
 
             usage = self.get_stage_param(stage, 'usage_', usage)
 
+            kpi_function_name = self.get_stage_param(stage, "kpi_function_name", kpi_function_name);
+
         except TypeError:
             is_executed = False
         else:
             is_executed = True
             if entities is not None or usage > 0:
-                self.trace_update(log_method=logger.debug, **{'entity_filter_list': entities, 'usage': usage})
+                self.trace_update(log_method=logger.debug, **{'entity_filter_list': entities, 'usage': usage, "kpi_function_name":kpi_function_name})
```

Comment:
Many functions (system function) wont have kpi function name. While executing this stage (Produce alert, inserting into database, data retrieval) we are adding "unknown", as an extra property that is not required.
It would be nice to add kpi_function_name if available. Else it will be just an extra unknown field in trace JSON.

Reply:
The trace update only gets called if there's an entity filter list or a nonzero usage value. Can system stages have usage values or entity filter lists? In testing I haven't seen that to be the case so far